gcloud/storage/acl.py

```diff
@@ -345,77 +348,64 @@ def get_entities(self):
         self._ensure_loaded()
         return list(self.entities.values())
 
-    def reload(self):
-        """Reload the ACL data from Cloud Storage.
+    @property
+    def reload_path(self):
+        """Compute the path for GET API requests for this ACL.
 
         This is a virtual method, expected to be implemented by subclasses.
 
-        :raises: :class:`NotImplementedError`
+        :raises: :class:`NotImplementedError` if ``_reload_path`` attribute
+                 is not set on the instance.
         """
+        # Allow override for testing
+        path = getattr(self, '_reload_path', None)
```

Comment:
The code you are objecting to *is* the second option -- it allows the base `ACL` class to have a "real" `reload_path` value for testing purposes:  we can't assign to the property (because it is read only), but we can assign to another attribute.  The only other alternative would be to remove the "pure virtual" bit:  maybe a `reload_property = None` assigned at class scope, which could be overridden for testing purposes on an instance.